src/app/core/google-analytics/google-analytics.effects.spec.ts

```diff
@@ -0,0 +1,49 @@
+import { getEffectsMetadata } from '@ngrx/effects';
+import { cold } from 'jasmine-marbles';
+import { GoogleAnalyticsEffects } from './google-analytics.effects';
+import { NavigationEnd } from '@angular/router';
+
+describe('GoogleAnalyticsEffects', () => {
+  let router: any;
+  const ga = (<any>window).ga;
+
+  beforeEach(() => {
+    router = {
+      routerState: {
+        snapshot: {}
+      },
+      events: {
+        pipe() {}
+      }
+    };
+
+    (<any>window).ga = jasmine.createSpy('ga');
+  });
+
+  afterAll(() => {
+    (<any>window).ga = ga;
+  });
+
+  it('should not dispatch action', function() {
+    const effect = new GoogleAnalyticsEffects(router);
+    const metadata = getEffectsMetadata(effect);
+
+    expect(metadata.pageView).toEqual({ dispatch: false });
+  });
+
+  it('should call google analytics', function() {
+    const routerEvent = new NavigationEnd(1, '', '');
+    router.events = cold('a', { a: routerEvent });
+    const effect = new GoogleAnalyticsEffects(router);
+
+    effect.pageView.subscribe(() => {
+      expect((<any>window).ga).toHaveBeenCalled();
```

Comment:
Isn't this test a bit redundant, because we're also testing it with `toHaveBeenCalledWith` ?

Reply:
Yes, that's true. I'll update that.